lib/protocol/SFTP.js

```diff
@@ -3695,8 +3695,10 @@ WriteStream.prototype._write = function(data, encoding, cb) {
                   this.pos,
                   (er, bytes) => {
     if (er) {
-      if (this.autoClose)
+      if (this.autoClose) {
+        this.emit('error', er); // er is swalloed by destroy(), emit manually.
         this.destroy();
```

Comment:
Can't we just instead do something like:

```suggestion
        this.destroy(er);
```

that should have the same effect.

Reply:
Yes, thank you. I've updated the branch and tested.